kpatch-build/gcc-plugins/ppc64le-plugin.c

```diff
@@ -46,8 +46,7 @@ static unsigned int ppc64le_plugin_execute(void)
 found:
 	if (nonlocal_code == -1 || local_code == -1 ||
 	    value_nonlocal_code == -1 || value_local_code == -1) {
-		fprintf(stderr, PLUGIN_NAME ": can't find call instruction codes");
-		return 1;
+		error("%s: cannot find call instruction codes", PLUGIN_NAME);
 	}
```

Comment:
Sorry for unclear commit message.  It does honor the return failure code from `ppc64le_plugin_execute`, it logs the error to `build.log` and skips converting the local calls to non-local for the current .o file and moves forward with kernel build.   `build.log` file would be flooded with `ppc64le-plugin: can't find call instruction codes` message, from every .o file, that encounters this failure but the kernel successfully.  The effect is of the failure is only seen while loading the kernel module.  This patch is more like hard stop of the kernel build using `error()` on the first instance of the issue.

Reply:
Hi @kamalesh-babulal : the change in behavior was clear enough, but without having any experience writing gcc plugins, I was wondering what the difference was between printing an error msg to stderr + returning non-zero vs. calling error().  Since you say it was previously flooding the build.log, I assumed that the caller of ppc64le_plugin_execute() did not consider non-zero return to be a fatal error.  So the fix is to call error() to force the build to terminate.

Reply:
Had a similar initial reflex wondering (not acquainted with gcc plugins) if this should be seen as a library where you generally don't want to call exit.

Reply:
@joe-lawrence true, that's my understanding too on the fatal vs non-fatal error @yhcote plugin is hooked into the compilation like an additional GCC pass and is invoked per compilation unit, returning 1 (failure) seems not to affect the other .o files.  One way I am aware to send fatal signal is error() and it better we bail out sooner.